zerver/views/streams.py

```diff
@@ -432,7 +433,9 @@ def add_subscriptions_backend(
             )
 
     if len(notifications) > 0:
-        do_send_messages(notifications)
+        message_ids = do_send_messages(notifications)
+        if len(created_streams) > 0:
+            mark_messages_as_read_by_stream_creator(user_profile, message_ids)
```

Comment:
I think we don't want to send a message and then mark it as read, since that will likely be user visible. 
I guess ideally `do_send_messages` could take a "mark as read" argument?

Reply:
A valid concern. But I'm not to sure if modifying `do_send_messages` is the best approach here either. It's not a bad idea, but I'm not sure if it's the best. @timabbott can we get your opinion on this as well?

Reply:
If we're going to do this concept at all, we should definitely have `do_send_messages` accept an argument to do this.  It's a much cleaner design.